### DISCUSSION TITLE
Avro code generation with Gradle multi-subproject build

### DISCUSSION THREAD

**Discussion by TheBroach** [quarkusio/quarkus#44193] [Q&A]
Hello,
I am working with Kafka and wanted to try out the Quarkus Apicurio Registry Avro extension. I am using Gradle and following their recommendation to structure the project as a multi-subproject build.

[Here is an MCVE](https://github.com/TheBroach/quarkus-gradle-avro-mcve). Give me a shout if something is missing from it!

As stated by the [official guide](https://quarkus.io/guides/kafka-schema-registry-avro#avro-schema), the `avsc` schema file needs to be located in `src/main/avro`, and when building, class files will be automatically generated from this schema file and put into the corresponding `build` folder. When working with a Gradle multi-subproject build, that would be the `src` and `build` folders of the Gradle subproject where the Quarkus Gradle plugin is applied, if I have understood things correctly. In the MCVE, that is the `application` subproject. 

However, as displayed by the MCVE, it would not be an unusual requirement to have to separate this out into a specific subproject, perhaps based on a design pattern or the desire to group different technical tools separately.

In the MCVE, since the `application` subproject is meant to assemble the Quarkus application it needs to specify dependencies on all other subprojects. But, for the `adapters` subproject to be able to use the generated classes, it would have to depend on the `application` subproject where they get generated, causing a circular dependency. Also, it would not be far fetched to wish for the `avsc` files to be located in the `adapters` subproject, where the resulting classes are going to be used. Is there any way I can achieve having both the schema file and the generated classes in the `adapters` subproject?

[a comment by an automated account is not shown]

**Reply by TheBroach** [reply]
Any thought on this @glefloch ? Or do you know anyone that you could tag that might have any ideas?

**Reply by TheBroach** [reply]
Maybe @cescoffier knows who to tag to give some insight on this?

**Reply by cescoffier** [reply]
Unfortunately, no - @glefloch would know the most about that part.

**Reply by TheBroach** [reply]
Will you have time soon to give this a thought @glefloch ? Or should I just abandon using code-generating Quarkus tasks when building multi-subproject Gradle applications where I do not want everything generated in the subproject where the Quarkus Gradle plugin is applied?
